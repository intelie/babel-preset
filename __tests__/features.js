import options from '../babel.config'
import { transformAsync } from '@babel/core'

describe('features', () => {
    test('optional chaining', async () => {
        const input = `const A = null; const B = A?.toFixed(2) || 3; B;`

        const { code: output } = await transformAsync(input, {
            ...options,
            filename: 'file.ts'
        })
        expect(output).toMatchSnapshot()

        const evaluated = eval(output)

        expect(evaluated).toEqual(3)
    })
    it('null coalesce', async () => {
        const input = `const A = null; A ?? 2`

        const { code: output } = await transformAsync(input, {
            ...options,
            filename: 'file.ts'
        })
        expect(output).toMatchSnapshot()

        const evaluated = eval(output)

        expect(evaluated).toEqual(2)
    })
    it('private methods', async () => {
        const input = `class B {

            #b () {
                return 42
            }

            getB () {
                return 42
            }

        }
        
        export default B
        `

        const { code: output } = await transformAsync(input, {
            ...options,
            filename: 'file.ts'
        })
        expect(output).toMatchSnapshot()

        const B = eval(output)
        const b = new B()

        expect(b.getB()).toEqual(42)
    })

    it('logical assignment', async () => {
        const input = `
        let a = 0
        const b = 2

        a ||= b
        
        export default a
        `

        const { code: output } = await transformAsync(input, {
            ...options,
            filename: 'file.ts'
        })
        expect(output).toMatchSnapshot()

        const a = eval(output)

        expect(a).toEqual(2)
    })
})
