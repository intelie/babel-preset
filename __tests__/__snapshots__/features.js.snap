// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`features logical assignment 1`] = `
"\\"use strict\\";

Object.defineProperty(exports, \\"__esModule\\", {
  value: true
});
exports.default = void 0;
var a = 0;
var b = 2;
a || (a = b);
var _default = a;
exports.default = _default;"
`;

exports[`features null coalesce 1`] = `
"\\"use strict\\";

var A = null;
A !== null && A !== void 0 ? A : 2;"
`;

exports[`features optional chaining 1`] = `
"\\"use strict\\";

var A = null;
var B = (A === null || A === void 0 ? void 0 : A.toFixed(2)) || 3;
B;"
`;

exports[`features private methods 1`] = `
"\\"use strict\\";

Object.defineProperty(exports, \\"__esModule\\", {
  value: true
});
exports.default = void 0;

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError(\\"Cannot call a class as a function\\"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if (\\"value\\" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

var _b = new WeakSet();

var B = /*#__PURE__*/function () {
  function B() {
    _classCallCheck(this, B);

    _b.add(this);
  }

  _createClass(B, [{
    key: \\"getB\\",
    value: function getB() {
      return 42;
    }
  }]);

  return B;
}();

var _b2 = function _b2() {
  return 42;
};

var _default = B;
exports.default = _default;"
`;
